### PR TITLE
Issue #2574513: doEvaluate for rules_list_contains

### PR DESCRIPTION
**Pull request by torenware**
Port of class DataListContains. Tests run green w/o modification.

Drupal issue: https://www.drupal.org/node/2574513

### CODE REVIEW

File: src/Plugin/Condition/DataListContains.php [hunk 1]
```diff
@@ -35,14 +35,11 @@
  * @todo: Add info alter
  */
 class DataListContains extends RulesConditionBase {
-
+
```
Comment (klausi) [inline src/Plugin/Condition/DataListContains.php]: trailing space

File: src/Plugin/Condition/DataListContains.php [hunk 2]
```diff
@@ -35,14 +35,11 @@
  * @todo: Add info alter
  */
 class DataListContains extends RulesConditionBase {
-
+  
   /**
-   * {@inheritdoc}
+   * Evaluate whether the list has the item
    */
```
Comment (klausi) [inline src/Plugin/Condition/DataListContains.php:46]: Please document the new params

File: src/Plugin/Condition/DataListContains.php [hunk 3]
```diff
@@ -52,8 +49,7 @@ public function evaluate() {
       }
       return FALSE;
     }
-
-    return in_array($item, $list);
+    return in_array($item, $list);
```
Comment (klausi) [inline src/Plugin/Condition/DataListContains.php]: Unrelated change that should be removed?

File: src/Plugin/Condition/DataListContains.php (hunk 2 above)
Comment (torenware) [inline src/Plugin/Condition/DataListContains.php:46]: I'm declaring $list as an array, since data type 'list' stores its data internally as an array.  Not sure in general how to translate some TypedData types into PHP types for this purpose.

File: src/Plugin/Condition/DataListContains.php (hunk 3 above)
Comment (torenware) [inline src/Plugin/Condition/DataListContains.php]: This looks like an area that varies a bit from plugin to plugin.  Similar plugins appear to put return statements flush with the "}" of the previous line.  This is not universal, but most plugins that have this kind of code structure omit the extra line.

If we want to enforce this, I guess we could always use a grunt plugin; must be one for this.

**Comment by torenware** [comment]
I've cleaned this up with drupalcs; param $list has been redeclared to better match the 'list' data type.

**Comment by klausi** [comment]
merged, thanks!
